src/main/java/org/mule/extension/http/api/request/HttpConnectivityValidator.java

```diff
@@ -0,0 +1,201 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.extension.http.api.request;
+
+import static java.lang.String.format;
+import static java.util.Collections.emptyMap;
+import static org.mule.runtime.api.util.MultiMap.emptyMultiMap;
+import static org.mule.runtime.extension.api.annotation.param.display.Placement.DEFAULT_TAB;
+import static org.slf4j.LoggerFactory.getLogger;
+
+import org.mule.extension.http.api.HttpResponseAttributes;
+import org.mule.extension.http.api.request.authentication.HttpRequestAuthentication;
+import org.mule.extension.http.api.request.authentication.UsernamePasswordAuthentication;
+import org.mule.extension.http.api.request.builder.HttpRequesterRequestBuilder;
+import org.mule.extension.http.api.request.validator.ResponseValidator;
+import org.mule.extension.http.api.request.validator.SuccessStatusCodeValidator;
+import org.mule.extension.http.internal.request.HttpRequesterCookieConfig;
+import org.mule.extension.http.internal.request.HttpResponseToResult;
+import org.mule.extension.http.internal.request.RequestConnectionParams;
+import org.mule.extension.http.internal.request.client.HttpExtensionClient;
+import org.mule.runtime.api.metadata.TypedValue;
+import org.mule.runtime.api.util.MultiMap;
+import org.mule.runtime.extension.api.annotation.param.Content;
+import org.mule.runtime.extension.api.annotation.param.NullSafe;
+import org.mule.runtime.extension.api.annotation.param.Optional;
+import org.mule.runtime.extension.api.annotation.param.Parameter;
+import org.mule.runtime.extension.api.annotation.param.display.DisplayName;
+import org.mule.runtime.extension.api.annotation.param.display.Placement;
+import org.mule.runtime.extension.api.runtime.operation.Result;
+import org.mule.runtime.http.api.client.auth.HttpAuthentication;
+import org.mule.runtime.http.api.domain.message.request.HttpRequest;
+import org.mule.runtime.http.api.domain.message.response.HttpResponse;
+import org.slf4j.Logger;
+
+import java.net.CookieManager;
+import java.util.Map;
+import java.util.concurrent.ExecutionException;
+
+public class HttpConnectivityValidator {
+
+  private static final Logger LOGGER = getLogger(HttpConnectivityValidator.class);
+
+  @Parameter
+  @Placement(tab = DEFAULT_TAB, order = 1)
+  private String path = "/";
+
+  /**
+   * HTTP Method for the request to be sent.
+   */
+  @Parameter
+  @Optional(defaultValue = "GET")
+  @Placement(order = 2)
+  private String method;
+
+  /**
+   * The body of the response message
+   */
+  @Parameter
+  @Content(primary = true)
+  @Placement(order = 3)
+  private TypedValue<Object> body;
```

Comment:
user cannot use DW expressions at this point. How is this body supposed to be built?

Reply:
Now it's a String. It seems to be enough given that it is used for connectivity testing